### ISSUE TITLE
No ENV declarations are passed into ssh daemon's shell

### ISSUE THREAD

**Issue by GunArm** [jenkinsci/docker-ssh-agent#46]
Possibly #33 is a symptom of this, but since #33 is more narrowly about JAVA functionality I thought it would be good to document this separately.

No dockerfile ENV settings are honored in the jenkins build environment (the ssh shell).  For instance I'd expect to be able to extend this image, put dotnet sdk on it, and opt out of  telemetry via ENV DOTNET_CLI_TELEMETRY_OPTOUT=true, but that would not be honored if set.  

I tried to work around this by using RUN to append to the /home/jenkins/.profile, but (I think?) that is blown away because jenkins dir is a volume.  I also tried putting it in /etc/profile, which works when I ssh in manually (as jenkins), but not in environment that results from jenkins plugin attaching.

**Comment by h1dden-da3m0n** [comment]
Hi, the issue you are describing is connected to the ssh connection Jenkins uses to access the agent.
There is an sshd setting `#PermitUserEnvironment no` which is causing this behaviour.

Possible solutions:
1. set all env-var values per agent via Jenkins
2. set all values globally on Jenkins
3. build a custom docker-ssh-agent image and sed `#PermitUserEnvironment no` to `PermitUserEnvironment yes`
4. not sure if this would be intended/accepted as a PR and request the addition of 3.) in each image to the [sed line](https://github.com/jenkinsci/docker-ssh-agent/blob/c8a81d44cc2b0ce9bde54c5c17b2bb201b5d3097/8/debian/buster/Dockerfile#L40), which is already present to change a couple of sshd options

hope this helps and was your problem 😉

**Comment by GunArm** [comment]
Apologies for having never replied to the above comment.  I did try the suggestions and they were not effective (but appreciated anyway) and then I got distracted going down a rabbit hole of further troubleshooting.  What ended up being the issue was tied to issue #62 and mitigated in PR #63 
In summary, I was running the container via docker plugin, which was subtley supplying /usr/sbin/sshd as a parameter (which docker plugin apparently does by default).  The start script, if given a command as a parameter, runs that command, and (prior to PR #63) does so before reaching the line that exports the variables.
